hooks/tk-multi-publish2/maya/collector_integ.py

```diff
@@ -18,13 +20,16 @@ def _resolve_item_fields(self, settings, item):
         # Now run the parent resolve method
         fields = super(MayaFocalLengthCollector, self)._resolve_item_fields(settings, item)
         publisher = self.parent
-        if item.properties.is_sequence:
-            first_frame = int(publisher.util.get_frame_number(item.properties.sequence_paths[0]))
-            end_frame = int(publisher.util.get_frame_number(item.properties.sequence_paths[-1]))
-            # TODO: Update to Dynamic process, When there is a need for different attribute values from the camera.
-            focal_length_dict = self.get_focal_length(first_frame, end_frame)
-            if focal_length_dict:
-                fields.update({"FocalLength": focal_length_dict})
+        all_cameras = cmds.listCameras()
+        if CAMERA_NAME in all_cameras:
+            if item.properties.is_sequence:
+                first_frame = int(publisher.util.get_frame_number(item.properties.sequence_paths[0]))
+                end_frame = int(publisher.util.get_frame_number(item.properties.sequence_paths[-1]))
```

Comment:
wouldn't first and the end frame be required, irrespective of the camera's presence?

Reply:
We use first and end frame to get the focallength values in all the frames from first to end.